apple/Sources/FerrostarMapLibreUI/Views/PortraitNavigationView.swift

```diff
@@ -7,63 +7,108 @@ import MapLibreSwiftUI
 import SwiftUI
 
 /// A portrait orientation navigation view that includes the InstructionsView at the top.
-public struct PortraitNavigationView: View {
+public struct PortraitNavigationView<TopCenter: View, TopTrailing: View, MidLeading: View, BottomTrailing: View>: View {
+    let theme: any FerrostarTheme
+
     let styleURL: URL
-    let distanceFormatter: Formatter
     // TODO: Configurable camera and user "puck" rotation modes
 
     private var navigationState: NavigationState?
     private let userLayers: [StyleLayerDefinition]
 
+    @ViewBuilder var topCenter: () -> TopCenter
+    @ViewBuilder var topTrailing: () -> TopTrailing
+    @ViewBuilder var midLeading: () -> MidLeading
+    @ViewBuilder var bottomTrailing: () -> BottomTrailing
+
     @Binding var camera: MapViewCamera
     @Binding var snappedZoom: Double
     @Binding var useSnappedCamera: Bool
 
+    var onTapExit: () -> Void
+
     public init(
+        theme: any FerrostarTheme = DefaultFerrostarTheme(),
         styleURL: URL,
         navigationState: NavigationState?,
         camera: Binding<MapViewCamera>,
         snappedZoom: Binding<Double>,
         useSnappedCamera: Binding<Bool>,
-        distanceFormatter: Formatter = MKDistanceFormatter(),
-        @MapViewContentBuilder _ makeMapContent: () -> [StyleLayerDefinition] = { [] }
+        onTapExit: @escaping () -> Void = {},
+        @MapViewContentBuilder mapContent: () -> [StyleLayerDefinition] = { [] },
+        @ViewBuilder topCenter: @escaping () -> TopCenter = { InfiniteSpacer() },
+        @ViewBuilder topTrailing: @escaping () -> TopTrailing = { InfiniteSpacer() },
+        @ViewBuilder midLeading: @escaping () -> MidLeading = { InfiniteSpacer() },
+        @ViewBuilder bottomTrailing: @escaping () -> BottomTrailing = { InfiniteSpacer() }
     ) {
+        self.theme = theme
         self.styleURL = styleURL
         self.navigationState = navigationState
-        self.distanceFormatter = distanceFormatter
-        userLayers = makeMapContent()
+        userLayers = mapContent()
```

Comment:
I'd kinda like to make this consistent. @ianthetechie did you happen to see any particular discussion if the way I did the @ViewBuilders is less ideal that what we you had here with the non-escaping capture in init?